Getting Started with JavaScript (#12689)
WordPress/gutenberg: docs/designers-developers/developers/tutorials/javascript/versions-and-building.md

> ## JavaScript versions and build step

Reviewer: this should be an h1 (`#`) with no newline at the top

> The Gutenberg Handbook shows JavaScript examples in two syntaxes ES5 and ESNext. These are version names for the JavaScript language standard definitions. You may also see elsewhere the versions ES6, or ECMAScript 2015 mentioned. Ses the [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) Wikipedia article for all the details.

Reviewer: ```suggestion
The Gutenberg Handbook shows JavaScript examples in two syntaxes: ES5 and ESNext. These are version names for the JavaScript language standard definitions. You may also see elsewhere the names ES6 or ECMAScript 2015 mentioned. See the [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) Wikipedia article for all the details.
```

ES5 code is compatible with WordPress's minimum [target for browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/).

For this tutorial, all examples are written compatible to ES5 definition of JavaScript. This code can be run straight in your browser and does not require an additional build step.

The ESNext version is dynamic and refers to the next language definitions, which will always be ahead of browser support. An extra build step is required to transform the code to a syntax that works in all browsers. Webpack and babel are the tools that perform this transformation step.

> Additionally, the ESNext code examples in the Gutenberg handbook include [JSX syntax](https://reactjs.org/docs/introducing-jsx.html), which is easier to read and write, but likewise requires the build step using webpack and babel to transform into compatible code.

Reviewer: These paragraphs should be reorganised a bit. Here's something to start from:

> ES5 code is compatible with WordPress's minimum [target for browser support](https://make.wordpress.org/core/handbook/best-practices/browser-support/).
>
> "ESNext" doesn't refer to a specific version of JavaScript, but is "dynamic" and refers to the next language definitions, whatever they might be. Because some browsers won't support these features yet (because they're new or proposed), an extra build step is required to transform the code to a syntax that works in all browsers. Webpack and babel are the tools that perform this transformation step.
>
> Additionally, the ESNext code examples in the Gutenberg handbook include [JSX syntax](https://reactjs.org/docs/introducing-jsx.html), a syntax that blends HTML and JavaScript. It makes it easier to read and write markup to be displayed in the browser, but likewise requires the build step using webpack and babel to transform into compatible code.
> 
> For simplicity, this tutorial uses the ES5 definition of JavaScript, without JSX. This code can be run straight in your browser and does not require an additional build step.

Author: 👍

> ## JavaScript versions and build step

Reviewer 2: This section is important but feels disconnected and left me wondered _how one should go about the build step?_. I know Gutenberg is agnostic about what tools third-party authors use, but I still feel this doc should direct them to the best resources available at the moment. It can be changed later if/when Gutenberg itself provides this. I'd include here a reference to the [Create Gutenberg Block](https://github.com/ahmadawais/create-guten-block) project.

Author: yeh, this was tricky. I wanted to explain what they might see in the documentation but try to keep it simple.  It is probably in the wrong spot in my table of contents, maybe put it after troubelshooting and expand.

I'm a big fan of getting the person to quick wins as they go through a tutorial.
Do this. Run. See something.
Now add this. Run. See new something.
-> Repeat.

So including the extra info in the middle confuses it. 
I think I'll reorganize it to the end and can expand.

The Gutenberg Handbook shows JavaScript examples in two syntaxes ES5 and ESNext. These are version names for the JavaScript language standard definitions. You may also see elsewhere the versions ES6, or ECMAScript 2015 mentioned. Ses the [ECMAScript](https://en.wikipedia.org/wiki/ECMAScript) Wikipedia article for all the details.

> ES5 code is compatible with WordPress's minimum [target for browser compatibility](https://make.wordpress.org/core/2017/04/23/target-browser-coverage/).

Reviewer 2: We may want to point to the [browser support handbook](https://make.wordpress.org/core/handbook/best-practices/browser-support/) or [the Gutenberg FAQ](https://wordpress.org/gutenberg/handbook/designers-developers/faq/#what-browsers-will-gutenberg-support) instead, they seem more permanent. I'd say the core handbook is probably best.

Author: 👍Updated the link.

For this tutorial, all examples are written compatible to ES5 definition of JavaScript. This code can be run straight in your browser and does not require an additional build step.

> The ESNext version is dynamic and refers to the next language defintions, which will always be ahead of browser support. An extra build step is required to transform the code to a syntax that works in all browsers. Webpack is the tool that performs this transformation step.

Reviewer 2: ```suggestion
The ESNext version is dynamic and refers to the next language definitions, which will always be ahead of browser support. An extra build step is required to transform the code to a syntax that works in all browsers. Webpack and babel are the tools that perform this transformation step.
```

> Additionally, the ESNext code examples in the Gutenberg handbook include JSX syntax, which is easier to read and write, but likewise requires the build step using webpack and babel to transform into compatible code.

Reviewer 2: ```suggestion
Additionally, the ESNext code examples in the Gutenberg handbook include [JSX syntax](https://reactjs.org/docs/introducing-jsx.html), which is easier to read and write, but likewise requires the build step using webpack and babel to transform into compatible code.
```
